File: 백준/Silver/10773. 제로/제로.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int k = Integer.parseInt(br.readLine());

        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < k; i++) {
            int num = Integer.parseInt(br.readLine());

            if (num == 0) {
                stack.pop();
            } else {
                stack.add(num);
            }
        }

        int result = 0;
        if (stack.isEmpty()) {
            System.out.println(0);
        } else {
            for (int s : stack) {
                result += s;
            }

            System.out.println(result);
        }
    }
}

File: 백준/Silver/10773. 제로/제로.py
import sys

k = int(sys.stdin.readline().strip())

stack = []

for _ in range(k):
    num = int(sys.stdin.readline().strip())
    
    if (num == 0):
        stack.pop()
    else:
        stack.append(num)
        
result = 0
if not stack:
    print(0)
else:
    print(sum(stack))